### PR TITLE
Add FC075 to detect node.save

### PR DESCRIPTION
**Pull request by tas50**
node.save is something you should only do if you REALLY know what you’re doing and are entirely sure actually you want a node.save to occur. Most people don’t need it and we really don’t want community cookbooks randomly saving the node (some do)

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -68,6 +68,10 @@ def resource_file(*args, &block)
       file("resources/my_resource.rb", *args, &block)
     end
 
+    def libary_file(*args, &block)
+      file("resources/helper.rb", *args, &block)
```
Comment (coderanger) [inline spec/spec_helper.rb]: That's not a library file?

File: spec/functional/fc075_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+require "spec_helper"
+
+describe "FC075" do
```
Comment (coderanger) [inline spec/functional/fc075_spec.rb]: Might be good to have a negative test for `node['save']` or whatevers.

File: lib/foodcritic/rules/fc075.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+rule "FC075", "Cookbook uses node.save to save data to the chef-server mid-run" do
```
Comment (coderanger) [inline lib/foodcritic/rules/fc075.rb]: Maybe "to save partial data" to make it clearer why this is bad?

**Comment by coderanger** [comment]
+1 for the idea, just need to fix the spec helper :)

File: spec/spec_helper.rb (hunk 1 above)
Comment (tas50) [inline spec/spec_helper.rb]: and I should be done for the night

**Pull request merged**